tests/components/template/test_sensor.py

```diff
@@ -114,6 +114,45 @@ def test_entity_picture_template(self):
         state = self.hass.states.get("sensor.test_template_sensor")
         assert state.attributes["entity_picture"] == "/local/sensor.png"
 
+    def test_available_template(self):
+        """Test available template."""
+        with assert_setup_component(1):
+            assert setup_component(
+                self.hass,
+                "sensor",
+                {
+                    "sensor": {
+                        "platform": "template",
+                        "sensors": {
+                            "test_template_sensor": {
+                                "value_template": "{{ states.sensor.test_state.state }}",
+                                "available_template": "{{ states.sensor.test_state.state == "
+                                "'Works' }}",
+                            }
+                        },
+                    }
+                },
+            )
+
+        self.hass.start()
+        self.hass.block_till_done()
+
+        state = self.hass.states.get("sensor.test_template_sensor")
+        assert not state.attributes.get("available")
+        assert state.state is None
```

Comment:
If it's not available, the state should be "unavailable"

https://github.com/home-assistant/home-assistant/blob/5994f82fc5d73822bd0516d7fb340c8a293105e0/homeassistant/helpers/entity.py#L297-L300